Airway - Add Manual Suction Pump

Apricot-ale opened the pull request:
**When merged this pull request will:**
- _Add `Manual Suction Pump` (almost same as ACCUVAC but it disposable and long use time). Disposable features can be optionally disabled or enabled. Of course, you can change the time._
- _Add "suction success hint" if suction is completed when there occluded._
- _Add display name string table for ACCUVAC (Item, Treatment menu)._
- _Change treatment action name for ACCUVAC. New name is `Airway Suction` at EN._
- _Change triage card treatment display name for ACCUVAC (show as `Airway Suction` at EN). This is treated as the same treatment display name as Manual Suction Pump._

Todo:
- [x] Add reusability option. (I don't know how... trying to refer IV reuse setting.)
- [x] Add dedicated hand pumping sound and suction sound.

Picture of item:
![suction_normal](https://github.com/KAT-Advanced-Medical/KAM/assets/50947830/cdc4dbe6-efd8-4381-9d2f-86fdf8557488)

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

Apricot-ale committed:
Add ManualSuctionPump, Add StringTable for Accuvac

Apricot-ale committed:
I found readme typo so fix and add MSP

it really necessary?

MiszczuZPolski commented on addons/airway/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -65,6 +65,17 @@ class ACE_Medical_Treatment_Actions {
         callbackProgress = "";
         sounds[] = {{QPATHTO_R(sounds\accuvac.wav),6,1,15}};
     };
+    class Suction: Larynxtubus {
```
i would inherit from accuvac class so you dont have to repeat those "methods" below that are same as accuvac

Apricot-ale replied:
Ok...
Now inheriting.
and I changed the order of class to reduce the number of lines of code as much as possible.

(don't want inherit sounds and consumeItem)

Apricot-ale replied:
it's solved

MiszczuZPolski commented on addons/airway/CfgWeapons.hpp:
```diff
@@ -38,4 +38,15 @@ class CfgWeapons {
     };
+    class kat_suction: ACE_ItemCore {
+        scope = 2;
+        author = "Katalam";
```
its not katalam tho 😄

Apricot-ale replied:
I forgot to change it. thank you.

Apricot-ale replied:
it's already solved too.

MiszczuZPolski commented:
> Add reusability option. (I don't know how)

look at consumeItem, but it depends on what you want to achieve

Apricot-ale committed:
change author

it's my work!

Apricot-ale committed:
Accuvac inherit Suction now

Reason of why order reversed:
because I don't want to set consumeItem = 1; again.

Apricot-ale committed:
Add new setting category "Suction Device"

Apricot-ale committed:
Add Reuse setting

I referred IV Reuse config.
Should I put it in AcuvacLocal?

Apricot-ale committed:
tab to space

Apricot-ale committed:
change == to isEqualTo

Apricot-ale committed:
Reuse setting strings and descriptions

Apricot-ale committed:
Change strings. Uppercase ACCUVAC

Change ACCUVAC item name to upper case
Remove ACCUVAC from the action name of ACCUVAC to make it concise.
Delete "pump" from the action name of "manual suction pump"

Apricot-ale committed:
Reduced usage time from 14s to 12s, Increased maximum time range to 30s

Apricot-ale committed:
Restore inheritance to Suction: Accuvac

Apricot-ale committed:
Replace todo with placeholder (old version accuvac sound)

Apricot-ale committed:
Update fnc_treatmentAdvanced_AccuvacStart.sqf

Add switching playSound3D for Manual or Accuvac

Apricot-ale committed:
fix misstypo naming treatment displayName

Apricot-ale committed:
change item name to lowercase for fix duplicate item on arsenal

Apricot-ale committed:
add 3den item version (cfgVehicles)

Apricot-ale committed:
Change file the doing re-add script for reuse

Apricot-ale committed:
Add hint for suction success

Apricot-ale committed:
fix "private" scope for sound playing

Apricot-ale committed:
Rename display name on triage card

Apricot-ale committed:
Fixed playback sound branching was not done correctly.

Apricot-ale committed:
Sound related changes

Added a special sound to the manual pump at start treatment.
Adjusted the tempo and length of the sound when a manual pump suction is successful (edited version old suction sound).
Rename Accuvac sounds to prevent confusion.

Apricot-ale committed:
Change ACCUVAC texts to uppercase.

Apricot-ale committed:
Remove redundant and unnecessary triage card treatment log distinctions

Apricot-ale committed:
Update manualsuction sound

add 1 sec blank

Apricot-ale committed:
no longer consumes items when medical suction not needed

Apricot-ale commented:
in game test complete

MissHeda commented on addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf:
```diff
@@ -25,8 +25,21 @@ params ["_medic", "_patient", "_bodyPart", "_classname", "", "_usedItem"];
     [_output, 1.5, _medic] call ACEFUNC(common,displayTextStructured);
+
+    if (_usedItem isEqualTo "kat_suction") then {
+        _medic addItem "kat_suction";
```
Why is here no check for reusable var as well?
If you put the sterile suction pump in the airway and use it but then notice "Ohh yea there is nothing, I should have checked airway first" it's not  sterile anymore. So in realism kind of sense you would not use it again on another patient.

Apricot-ale replied:
I never thought of that idea. thanks.

MissHeda commented on addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf:
```diff
@@ -25,8 +25,21 @@ params ["_medic", "_patient", "_bodyPart", "_classname", "", "_usedItem"];
 };
 
-[_patient, _classname] call ACEFUNC(medical_treatment,addToTriageCard);
+[_patient, "Airway Suction"] call ACEFUNC(medical_treatment,addToTriageCard);
```
What if someone is playing in like German and does not understand what this means?
Please make universal via a string or use classname as before so everybody understands if language differs from English.

Apricot-ale replied:
The class name looks good, so delete that line.

Apricot-ale replied:
By the way, did you know that if you use a class name instead of English, it will be written to triagecard as "ACCUVAC" and "SUCTION" in any language?
Even in other medical procedures, most of the information is written in English or by class name.
Or do need to go to the trouble of using German and Japanese or some languages for only airway suctions?

I chose "Airway Suction" to show in the log that ACCUVAC and SUCTION are doing the same thing (Airway Suction).
Did you want the log to show ACCUVAC and SUCTION separately?

as classname in not English langage config:
![20240125064752_1](https://github.com/KAT-Advanced-Medical/KAM/assets/50947830/4c23c5a5-18f0-471a-971b-e19955a01d28)

as "airway suction" in not English langage config:
![20240125065409_1](https://github.com/KAT-Advanced-Medical/KAM/assets/50947830/c2a1e8f0-c8e1-49b3-b3c6-6832f8f3688e)

Apricot-ale replied:
Other examples of using English instead of class names https://github.com/KAT-Advanced-Medical/KAM/blob/78669c517b6915b7887f5a9766f7d2d1e420e9d2/addons/pharma/functions/fnc_treatmentAdvanced_DialysisLocal.sqf#L27

Apricot-ale replied:
https://github.com/KAT-Advanced-Medical/KAM/pull/453/commits/8cc4c0475d19507dc266394bd7e7bc788acb60de
This change looks ok.

Apricot-ale committed:
add reuse check for failed for accuvac

Apricot-ale committed:
revert to classname

Apricot-ale committed:
Merge branch 'KAT-Advanced-Medical:dev-Tomcat' into ManualSuctionDevice

Apricot-ale committed:
Revert "revert to classname"

This reverts commit 7de2c94addcc62fb90c18cae2ab6af65627ef4ae.

Apricot-ale committed:
Triagecard treatmentlog localize

Apricot-ale committed:
merge "remove arsenal" change

Apricot-ale committed:
Merge branch 'KAT-Advanced-Medical:dev-Tomcat' into ManualSuctionDevice

Apricot-ale committed:
remove comment and hidden original line

MissHeda reviewed (approved):
Looks good to me.

MissHeda commented on addons/airway/CfgWeapons.hpp:
```diff
@@ -38,4 +38,15 @@ class CfgWeapons {
+        picture = QPATHTOF(ui\suction_normal.paa);
+        ACE_isMedicalItem = 1;
+        class ItemInfo: CBA_MiscItem_ItemInfo {
+            mass = 1.5;
```
change to a lower more realistic number. 1.5 is quite big.

Apricot-ale replied:
changed

Apricot-ale committed:
make small suction device inventory size.

MissHeda reviewed (approved):
Everything matches out, @MiszczuZPolski ready to merge now.

MiszczuZPolski commented on addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf:
```diff
@@ -25,8 +25,24 @@ params ["_medic", "_patient", "_bodyPart", "_classname", "", "_usedItem"];
     [_output, 1.5, _medic] call ACEFUNC(common,displayTextStructured);
+
+    if (_usedItem isEqualTo "kat_suction") then {
+        if(GVAR(Suction_reuse)) then {
```
```suggestion
        if (GVAR(Suction_reuse)) then {
```

MiszczuZPolski commented on addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf:
```diff
@@ -25,8 +25,24 @@ params ["_medic", "_patient", "_bodyPart", "_classname", "", "_usedItem"];
+};
+
+if (_usedItem isEqualTo "kat_suction") then {
+    if(GVAR(Suction_reuse)) then {
```
```suggestion
    if (GVAR(Suction_reuse)) then {
```

MiszczuZPolski committed:
Update addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf

MiszczuZPolski committed:
Update addons/airway/functions/fnc_treatmentAdvanced_Accuvac.sqf

MiszczuZPolski reviewed (approved):
LGTM
